pkg/gather/ssh/agent.go

```diff
@@ -0,0 +1,36 @@
+package ssh
+
+import (
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/ssh/agent"
+
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+)
+
+// newAgent initializes an SSH Agent with the keys.
+// If not keys are provided, it loads all the keys from the user's environment.
+func newAgent(keyPaths []string) (agent.Agent, error) {
+	keys, err := loadKeys(keyPaths)
+	if err != nil {
+		return nil, err
+	}
+
+	ag := agent.NewKeyring()
+	var errs []error
+	for idx := range keys {
+		if err := ag.Add(agent.AddedKey{PrivateKey: keys[idx]}); err != nil {
```

Comment:
nit:

```go
for _, key := range keys {
  if err := ag.Add(agent.AddedKey{PrivateKey: key}); err != nil {
```

or do you need to use `keys[idx]` for some Go-closure reason that I'm forgetting about?

Reply:
well both ways are acceptable in Go.. do you have any reason to prefer the `_, obj` over `idx`?

Reply:
> do you have any reason to prefer the `_, obj` over `idx`?

It saves us an a later array lookup, makes it clear that we aren't doing anything fancy with the index, and seems to be more common in our existing usage:

```console
$ git describe
unreleased-master-1105-g0e4119dd1
$ git grep 'for _, .* := range' pkg | grep -v pkg/terraform/exec | wc -l
176
$ git grep 'for [^,]* := range' pkg | grep -v pkg/terraform/exec | wc -l
16
```

I don't see anything in [the style guide][1] about this, and [Effective Go only says][2]:

> If you only need the second item in the range (the value), use the *blank identifier*, an underscore, to discard the first:

which could be "you should do it this way" or could be "you can do it this way".  So if you're more comfortable getting the index and performing a later array lookup, I'll survive ;).

[1]: https://github.com/golang/go/wiki/CodeReviewComments
[2]: https://golang.org/doc/effective_go.html#for